### PR TITLE
Run-time configuration of site modes

### PR DESCRIPTION
**Pull request by dpoirier**
Add a model with one record. Use admin to set its fields
to set:

* whether to show vote counts
* whether to show total votes
* whether to allow sorting questions by votes
* the date/time of the debate

### CODE REVIEW

**Comment by ejucovy** [comment]
Very cool!

Vote tallies still sneak in after casting a vote -- we'll need to at least adjust https://github.com/caktus/django-opendebates/blob/sitemodes/opendebates/static/js/base/helpers.js#L74 to prevent this, and preferably also the backend somehow (https://github.com/caktus/django-opendebates/blob/sitemodes/opendebates/views.py#L246) so that tech-savvy users can't get at vote tallies by looking in developer tools.

**Comment by dpoirier** [comment]
Thanks @ejucovy, I'll take a look.

**Comment by dpoirier** [comment]
Closed loophole.

File: opendebates/models.py [hunk 1]
```diff
@@ -28,6 +28,19 @@ class Meta:
...
 
 
+class SiteMode(CachingMixin, models.Model):
+    show_question_votes = models.BooleanField(default=True, blank=True)
+    show_total_votes = models.BooleanField(default=True, blank=True)
+    allow_sorting_by_votes = models.BooleanField(default=True, blank=True)
+    allow_voting_and_submitting_questions = models.BooleanField(default=True, blank=True)
+    debate_time = models.DateTimeField(
+        default=datetime.datetime(2099, 1, 1),
+        help_text="Enter time that debate starts in timezone %s" % settings.TIME_ZONE,
```
Comment (vkurup) [inline opendebates/models.py]: Having Model `help_text` vary based on a setting means that Django will require a migration anytime the setting is changed. I don't think that's an issue here, but just wanted to mention it.

**Comment by vkurup** [comment]
When I disable question submission, the placeholder text in the search box still says something about 'please search before submitting a question'. We could maybe change that text so it leaves out the 'submitting a question' part.

Then :+1:

**Pull request merged**